### PR TITLE
UV photos - Wish from LarsHP Increased green (tint) value from 10 to 100

### LINKED ISSUE #7044: UV photos - Wish from LarsHP Increased green (tint) value from 10 to 40
https://discuss.pixls.us/t/extreme-camera-white-balance-is-off-in-rawtherapee/43193

I create a Pull-Request 
https://github.com/Beep6581/RawTherapee/pull/7043

I made a comment with the PR, so as not to multiply the places for discussion.
I open this issue to respect the usual rules.

Jacques

### PR DESCRIPTION
**Pull request by Desmis**
In order to allow better white balance settings when the illuminant - generally filtered - leads to shots where UV or IR are dominant, the usual settings are insufficient.

At the request of LarsHP I opened this PR.

It increases the extreme possibilities for the tint which is now from 0.02 to 40. Standard value = 1.
The temperature is not changed from 1500K to 60000K. Usual values between 2800K to 8000K.

As a reminder, there is a mathematical link between channel multipliers and “Temperature and tint”.

In 99% of cases we use "Temp" and "Tint" because on the one hand it is more intuitive and on the other hand it is what is included in standards and habits. We are talking about the illuminant D50, D65, Tungsten (stdA)... Even the novice user knows the concepts of Temperature. Ciecam use only "Temperature". 

For information, the human eye is really sensitive from 380nm to 740nm. Ciexy is from 350nm to 830nm.
The UV zone is between 100 and 400nm
The IR zone is between 780 and 830nm and above.

Internally at RT (matrix calculation), the illuminants (D50, stdA, F11, LED...) are measured from 350nm to 830nm
Colors (flowers, minerals, nature...) are rated from 380nm to 740nm
Observer 2° or 10° from 360nm to 830nm

Jacques

### CODE REVIEW

**Comment by Desmis** [comment]
Exemple 
//spectral data Colorchecker24 : Red C3 - from 350nm to 830nm (step 5nm)
const double  ColorTemp::ColorchechredC3_spect[97] = {
    0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
    0.0478, 0.0476, 0.0474, 0.0472, 0.0470, 0.0466, 0.0461, 0.0460, 0.0459, 0.0456, 0.0453, 0.0451, 0.0449, 0.0448, 0.0447, 0.0446, 0.0445, 0.0441, 0.0437, 0.0432, 0.0427, 0.0424, 0.0421, 0.0419,
    0.0417, 0.0415, 0.0412, 0.0412, 0.0412, 0.0413, 0.0413, 0.0415, 0.0416, 0.0421, 0.0426, 0.0436, 0.0446, 0.0469, 0.0491, 0.0549, 0.0607, 0.0773, 0.0939, 0.1376, 0.1812, 0.2568, 0.3323, 0.4070, 0.4816, 0.5308,
    0.5800, 0.6059, 0.6317, 0.6447, 0.6577, 0.6653, 0.6728, 0.6761, 0.6793, 0.6820, 0.6847, 0.6878, 0.6909, 0.6945, 0.6980, 0.7013, 0.7046, 0.7065, 0.7084, 0.7107, 0.7129, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
    0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0
};

illuminant D5300 from 350nm to 830nm
const double ColorTemp::Daylight5300_spect[97] = {
    24.82, 26.27, 27.72, 28.97, 30.22, 29.71, 29.19, 31.95, 34.71, 45.49, 56.26, 59.97, 63.68, 65.30, 66.92, 65.39, 63.86, 72.59, 81.32, 87.53, 93.73, 95.15, 96.56, 96.55, 96.54, 98.13, 99.73, 97.70, 95.66, 97.19, 98.72,
    98.90, 99.08, 98.98, 98.87, 101.13, 103.39, 102.48, 101.57, 102.14, 102.71, 101.36, 100.00, 98.71, 97.42, 97.81, 98.21, 95.20, 92.20, 93.92, 95.63, 96.15, 96.67, 96.34, 96.01, 94.21, 92.41, 93.58, 94.74, 93.05, 91.36, 92.29,
    93.21, 95.25, 97.28, 95.30, 93.32, 87.92, 82.51, 84.29, 86.06, 86.94, 87.81, 80.24, 72.68, 77.32, 81.96, 84.88, 87.79, 81.01, 74.22, 64.41, 54.60, 66.55, 78.51, 76.35, 74.20, 74.79, 75.38, 72.48, 69.58, 65.11, 60.64,
    63.88, 67.13, 68.85, 70.57
};

The different amplitude, between the two examples, does not matter

**Comment by Desmis** [comment]
And Observer 2° from 350nm to 830 nm for XYZ values, "human sight"
static const color_match_type cie_colour_match_jd2 = {//350nm to 830nm   5 nm J.Desmis 2° Standard Observer.
    {0.0000000, 0.000000, 0.000000}, {0.0000000, 0.000000, 0.000000}, {0.0001299, 0.0003917, 0.0006061}, {0.0002321, 0.000006965, 0.001086}, {0.0004149, 0.00001239, 0.001946}, {0.0007416, 0.00002202, 0.003846},
    {0.001368, 0.000039, 0.006450001}, {0.002236, 0.000064, 0.01054999}, {0.004243, 0.000120, 0.02005001}, {0.007650, 0.000217, 0.036210}, {0.014310, 0.000396, 0.06785001}, {0.023190, 0.000640, 0.110200},
    {0.043510, 0.001210, 0.207400}, {0.077630, 0.002180, 0.371300}, {0.134380, 0.004000, 0.645600}, {0.214770, 0.007300, 1.0390501}, {0.283900, 0.011600, 1.385600}, {0.328500, 0.016840, 1.622960},
    {0.348280, 0.023000, 1.747060}, {0.348060, 0.029800, 1.782600}, {0.336200, 0.038000, 1.772110}, {0.318700, 0.048000, 1.744100}, {0.290800, 0.060000, 1.669200}, {0.251100, 0.073900, 1.528100},
    {0.195360, 0.090980, 1.287640}, {0.142100, 0.112600, 1.041900}, {0.095640, 0.139020, 0.8129501}, {0.05795001, 0.169300, 0.616200}, {0.032010, 0.208020, 0.465180}, {0.014700, 0.258600, 0.353300},
    {0.004900, 0.323000, 0.272000}, {0.002400, 0.407300, 0.212300}, {0.009300, 0.503000, 0.158200}, {0.029100, 0.608200, 0.111700}, {0.063270, 0.710000, 0.07824999}, {0.109600, 0.793200, 0.05725001},
    {0.165500, 0.862000, 0.042160}, {0.2257499, 0.9148501, 0.029840}, {0.290400, 0.954000, 0.020300}, {0.359700, 0.980300, 0.013400}, {0.43344990, 0.9949501, 0.008749999}, {0.5120501, 1.000000, 0.005749999},
    {0.594500, 0.995000, 0.003900}, {0.678400, 0.978600, 0.002749999}, {0.762100, 0.952000, 0.002100}, {0.842500, 0.915400, 0.001800}, {0.916300, 0.870000, 0.001650001}, {0.978600, 0.816300, 0.001400},
    {1.026300, 0.757000, 0.001100}, {1.056700, 0.694900, 0.001000}, {1.062200, 0.631000, 0.000800}, {1.045600, 0.566800, 0.000600}, 
    {1.002600, 0.503000, 0.000340}, {0.938400, 0.441200, 0.000240}, {0.8544499, 0.381000, 0.000190}, {0.751400, 0.321000, 0.000100}, {0.642400, 0.265000, 0.00004999999},
    {0.541900, 0.217000, 0.000030}, {0.447900, 0.175000, 0.000020}, {0.360800, 0.138200, 0.000010}, {0.283500, 0.107000, 0.000000}, {0.218700, 0.081600, 0.000000}, {0.164900, 0.061000, 0.000000},
    {0.121200, 0.044580, 0.000000}, {0.087400, 0.032000, 0.000000}, {0.063600, 0.023200, 0.000000}, {0.046770, 0.017000, 0.000000}, {0.032900, 0.011920, 0.000000}, {0.022700, 0.008210, 0.000000},
    {0.015840, 0.005723, 0.000000}, {0.01135916, 0.004102, 0.000000}, {0.008110916, 0.002929, 0.000000}, {0.005790346, 0.002091, 0.000000}, {0.004109457, 0.001484, 0.000000}, {0.002899327, 0.001047, 0.000000},
    {0.00204919, 0.000740, 0.000000}, {0.001439971, 0.000520, 0.000000}, {0.0009999493, 0.0003611, 0.000000}, {0.0006900786, 0.0002492, 0.000000}, {0.0004760213, 0.0001719, 0.000000}, {0.0003323011, 0.000120, 0.000000},
    {0.0002348261, 0.0000848, 0.000000}, {0.0001661505, 0.000060, 0.000000}, {0.000117413, 0.0000424, 0.000000},{0.00008307527, 0.000030, 0.000000}, {0.00005870652, 0.0000212, 0.000000}, {0.00004150994, 0.00001499, 0.000000},
    {0.00002935326, 0.0000106, 0.000000}, {0.00002067383, 0.0000074657, 0.000000}, {0.00001455977, 0.0000052578, 0.000000}, {0.00001025398, 0.0000037029, 0.000000}, {0.000007221456, 0.00000260778, 0.000000}, {0.000005085868, 0.0000018366, 0.000000},
    {0.000003581652, 0.0000012934, 0.000000}, {0.000002522525, 0.00000091093, 0.000000}, {0.000001776509, 0.00000064153, 0.000000}, {0.000001251141, 0.00000045181, 0.000000}
};

Note the values 0. for x and y, and z for 350nm

File: rtgui/whitebalance.cc [hunk 1]
```diff
@@ -396,6 +396,7 @@ WhiteBalance::WhiteBalance () : FoldableToolPanel(this, TOOL_NAME, M("TP_WBALANC
     pack_start(*StudLabel);
     pack_start(*PatchLabel);
     pack_start(*PatchlevelLabel);
+    green->setLogScale(100, 0.);
```
Comment (Lawrence37) [inline rtgui/whitebalance.cc]: The green slider had always been hard to adjust. I prefer the value 1.0 to be near the center with an even logarithmic scale. I think `setLogScale(MAXGREEN / MINGREEN, MINGREEN)` will do something similar, but I have not tried.

Comment (Desmis) [reply rtgui/whitebalance.cc]: Yes, It's better that way, more intuitive. I push for change
green->setLogScale(MAXGREEN / MINGREEN, MINGREEN)

**Comment by Desmis** [comment]
@Lawrence37 

If no object, I will merge tomorrow :)

Jacques

**Pull request merged**